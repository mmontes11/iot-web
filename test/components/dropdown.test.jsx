import React from "react";
import { shallow } from "enzyme";
import Dropdown from "components/dropdown";

describe("components/dropdown", () => {
  it("renders in initial state", () => {
    const wrapper = shallow(
      <Dropdown
        label="Select item"
        items={["foo", "bar"]}
        isActive={false}
        isLoading={false}
        isDisabled={false}
        onButtonClick={() => undefined}
        onItemClick={() => undefined}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders in active state", () => {
    const wrapper = shallow(
      <Dropdown
        label="Select item"
        items={["foo", "bar"]}
        isActive
        isLoading={false}
        isDisabled={false}
        onButtonClick={() => undefined}
        onItemClick={() => undefined}
      />,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".dropdown").hasClass("is-active")).toBeTruthy();
  });
  it("renders in loading state", () => {
    const wrapper = shallow(
      <Dropdown
        label="Select item"
        items={["foo", "bar"]}
        isActive={false}
        isLoading
        isDisabled={false}
        onButtonClick={() => undefined}
        onItemClick={() => undefined}
      />,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".button").hasClass("is-loading")).toBeTruthy();
  });
  it("renders in disabled state", () => {
    const wrapper = shallow(
      <Dropdown
        label="Select item"
        items={["foo", "bar"]}
        isActive={false}
        isLoading={false}
        isDisabled
        onButtonClick={() => undefined}
        onItemClick={() => undefined}
      />,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".button").prop("disabled")).toBeTruthy();
  });
  it("simulates a click in button", () => {
    const onButtonClick = jest.fn();
    const wrapper = shallow(
      <Dropdown
        label="Select item"
        items={["foo", "bar"]}
        isActive={false}
        isLoading={false}
        isDisabled={false}
        onButtonClick={onButtonClick}
        onItemClick={() => undefined}
      />,
    );
    wrapper.find(".button").simulate("click");
    expect(onButtonClick).toHaveBeenCalledTimes(1);
  });
  it("simulates a click in an item", () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <Dropdown
        label="Select item"
        items={["foo", "bar"]}
        isActive={false}
        isLoading={false}
        isDisabled={false}
        onButtonClick={onClick}
        onItemClick={onClick}
      />,
    );
    const dropdownItem = wrapper.find(".dropdown-item").first();
    dropdownItem.simulate("click");
    dropdownItem.simulate("keypress", { key: "a" });
    expect(onClick).toHaveBeenCalledTimes(2);
  });
});
