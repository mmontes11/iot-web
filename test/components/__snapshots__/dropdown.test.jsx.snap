// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/dropdown renders in active state 1`] = `
<div
  className="dropdown is-active"
>
  <div
    className="dropdown-trigger"
  >
    <button
      aria-controls="dropdown-menu"
      aria-haspopup="true"
      className="button"
      disabled={false}
      onClick={[Function]}
    >
      <span>
        Select item
      </span>
      <span
        className="icon is-small"
      >
        <i
          aria-hidden="true"
          className="fas fa-angle-down"
        />
      </span>
    </button>
  </div>
  <div
    className="dropdown-menu"
    id="dropdown-menu"
    role="menu"
  >
    <div
      className="dropdown-content"
    >
      <div
        className="dropdown-item"
        key="foo"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        foo
      </div>
      <div
        className="dropdown-item"
        key="bar"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        bar
      </div>
    </div>
  </div>
</div>
`;

exports[`components/dropdown renders in disabled state 1`] = `
<div
  className="dropdown"
>
  <div
    className="dropdown-trigger"
  >
    <button
      aria-controls="dropdown-menu"
      aria-haspopup="true"
      className="button"
      disabled={true}
      onClick={[Function]}
    >
      <span>
        Select item
      </span>
      <span
        className="icon is-small"
      >
        <i
          aria-hidden="true"
          className="fas fa-angle-down"
        />
      </span>
    </button>
  </div>
  <div
    className="dropdown-menu"
    id="dropdown-menu"
    role="menu"
  >
    <div
      className="dropdown-content"
    >
      <div
        className="dropdown-item"
        key="foo"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        foo
      </div>
      <div
        className="dropdown-item"
        key="bar"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        bar
      </div>
    </div>
  </div>
</div>
`;

exports[`components/dropdown renders in initial state 1`] = `
<div
  className="dropdown"
>
  <div
    className="dropdown-trigger"
  >
    <button
      aria-controls="dropdown-menu"
      aria-haspopup="true"
      className="button"
      disabled={false}
      onClick={[Function]}
    >
      <span>
        Select item
      </span>
      <span
        className="icon is-small"
      >
        <i
          aria-hidden="true"
          className="fas fa-angle-down"
        />
      </span>
    </button>
  </div>
  <div
    className="dropdown-menu"
    id="dropdown-menu"
    role="menu"
  >
    <div
      className="dropdown-content"
    >
      <div
        className="dropdown-item"
        key="foo"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        foo
      </div>
      <div
        className="dropdown-item"
        key="bar"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        bar
      </div>
    </div>
  </div>
</div>
`;

exports[`components/dropdown renders in loading state 1`] = `
<div
  className="dropdown"
>
  <div
    className="dropdown-trigger"
  >
    <button
      aria-controls="dropdown-menu"
      aria-haspopup="true"
      className="button is-loading"
      disabled={false}
      onClick={[Function]}
    >
      <span>
        Select item
      </span>
      <span
        className="icon is-small"
      >
        <i
          aria-hidden="true"
          className="fas fa-angle-down"
        />
      </span>
    </button>
  </div>
  <div
    className="dropdown-menu"
    id="dropdown-menu"
    role="menu"
  >
    <div
      className="dropdown-content"
    >
      <div
        className="dropdown-item"
        key="foo"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        foo
      </div>
      <div
        className="dropdown-item"
        key="bar"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        bar
      </div>
    </div>
  </div>
</div>
`;
